src/core/Proc/Async.pm6

```diff
@@ -412,15 +424,24 @@ my class Proc::Async {
     proto method kill(|) {*}
```

Comment:
Might be worth consolidating all the duplicated logic (including some existing) into the proto:
```
proto method kill(|) {
    X::Proc::Async::MustBeStarted.new(:method<kill>, proc => self).throw if !$!started;
    $!stdout_supplier.done() if $!stdout_supplier;
    $!stderr_supplier.done() if $!stderr_supplier;
    $!merge_supplier.done()  if $!merge_supplier;
    {*}
}
```

Reply:
I had left the duplicate logic because of the comment a few lines above stating that it significantly impacted performance. That comment is quite old though so things could be different now with the improvements made to the optimizer and spesh.